packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -321,6 +322,23 @@ class FlutterPlugin implements Plugin<Project> {
         String flutterExecutableName = Os.isFamily(Os.FAMILY_WINDOWS) ? "flutter.bat" : "flutter"
         flutterExecutable = Paths.get(flutterRoot.absolutePath, "bin", flutterExecutableName).toFile()
 
+        // Validate that the provided Gradle, Java, AGP, and KGP versions are all within our
+        // supported range.
+        final Boolean shouldSkipDependencyChecks = project.hasProperty("skipDependencyChecks")
+                && project.getProperty("skipDependencyChecks");
+        if (!shouldSkipDependencyChecks) {
+            try {
+                final String dependencyCheckerPluginPath = Paths.get(flutterRoot.absolutePath,
+                        "packages", "flutter_tools", "gradle", "src", "main", "kotlin",
+                        "dependency_version_checker.gradle.kts")
+                project.apply from: dependencyCheckerPluginPath
```

Comment:
maybe we could add a `TODO` or something similar that links to https://github.com/flutter/flutter/issues/121541. E.g.

```
We apply a Gradle plugin because we can't use import statements for backward compatibility reasons. See #121541 for more info.
```

just an idea

Reply:
or link to this comment: https://github.com/flutter/flutter/pull/142000#discussion_r1465560136

(I can image people in the future being confused by this)

Reply:
Good idea, added a todo with explanation and link to https://github.com/flutter/flutter/issues/121541#issuecomment-1920363687